### PR TITLE
Big changes to cohort corruption, TSF, and NPQEYL marts

### PR DESCRIPTION
**Pull request by emiliocampa**
**ls_cohort_corruption_checker:** this is an old mart that has been resurrected. The code for this had been entirely swapped out with the very complex cohort checker code that I've previously been running for Nathan. It has a lot of complex policy checks in it, and the table serves the joint purpose of showing all those declarations that don't align with a participant's cohort and also then providing all declarations for those participants.

**ls_npq_tsf_details:** this is a new mart. This is the Targeted Support Funding table that powers the NPQTSF dashboard. The code itself isn't that complex.

**ls_npqeyl_enrolments**: this is an update to definitions. Following discussions with the NPQEYL team, we have agreed upon a set of definitions for what is captured under "setting type". This is quite complex and there will be a section in the NPQEYL dashboard that lays out the agreed-upon definitions more clearly.

I'm happy to talk through this if you have questions.

### CODE REVIEW

File: definitions/marts/looker_studio_marts/ecf/ls_cohort_corruption_checker.sqlx [hunk 1]
```diff
@@ -1,60 +1,235 @@
...
-        induction_record_id: "ID of individual induction record"
+        id: "ID of declaration"
     }
 }
 
+WITH
+  full_cohort_checker_output AS (
+  WITH
+    full_declaration_milestone_table AS (
+    /*This section creates a list of all s/e/p/p declarations for a participant, pulls in details from their current induction record, and then joins on the two static tables ecf_schedule_milestone_mapping and ecf_schedule_milestone_mapping_condensed to later make comparisons against the declaration_date of each participant*/
```
Comment (consult1815) [inline definitions/marts/looker_studio_marts/ecf/ls_cohort_corruption_checker.sqlx:20]: What is s/e/p/p

File: definitions/marts/looker_studio_marts/ecf/ls_cohort_corruption_checker.sqlx [hunk 2]
```diff
@@ -1,60 +1,235 @@
...
+      AS suggested_cohort_from_declaration
+    FROM
+      full_declaration_milestone_table fdmt
+    LEFT JOIN
+      statement_information si
+    ON
+      fdmt.declaration_id=si.participant_declaration_id)
+  SELECT
+    *,
+    CASE
```
Comment (consult1815) [inline definitions/marts/looker_studio_marts/ecf/ls_cohort_corruption_checker.sqlx:139]: Can you add a note here about what this case statement is doing?

File: definitions/marts/looker_studio_marts/ecf/ls_cohort_corruption_checker.sqlx [hunk 3]
```diff
@@ -1,60 +1,235 @@
...
+  c.suggested_cohort_from_declaration,
+  c.check_source
 FROM
-  ${ref(`ecf_golden_thread`)}
+  `dataform.ls_declarations_provider_names` a
+INNER JOIN
+  full_cohort_checker_output b
+ON
+  a.participant_profile_id = b.participant_profile_id
+LEFT JOIN
```
Comment (consult1815) [inline definitions/marts/looker_studio_marts/ecf/ls_cohort_corruption_checker.sqlx:230]: Is this to see declarations that are correct and those that are incorrect?

Comment (consult1815) [reply definitions/marts/looker_studio_marts/ecf/ls_cohort_corruption_checker.sqlx:230]: vs. inner join which would just be the incorrect ones?

File: definitions/marts/looker_studio_marts/npq/ls_npq_tsf_details.sqlx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    WHEN gias.CloseDate IS NOT NULL THEN 0
+END
+  AS sch_open
+FROM
+  valid_start_npq_apps npq
+LEFT JOIN
+  gias_information gias
+ON
+  npq.school_urn = gias.string_urn
+WHERE
```
Comment (consult1815) [inline definitions/marts/looker_studio_marts/npq/ls_npq_tsf_details.sqlx:85]: Make this null?

**Review by consult1815 — CHANGES_REQUESTED** [review]
Will run a SQL check on the cohort checker when changes have been made.

File: definitions/marts/looker_studio_marts/ecf/ls_cohort_corruption_checker.sqlx (hunk 1 above)
Comment (emiliocampa) [reply definitions/marts/looker_studio_marts/ecf/ls_cohort_corruption_checker.sqlx:20]: submitted eligible payable paid

File: definitions/marts/looker_studio_marts/ecf/ls_cohort_corruption_checker.sqlx [hunk 5]
```diff
@@ -1,60 +1,235 @@
...
...stone_date)) OR (declaration_type = 'retained-1' AND (dec_date_short NOT BETWEEN ret1_dec_start_date AND ret1_dec_milestone_date)) OR (declaration_type = 'retained-2' AND (dec_date_short NOT BETWEEN ret2_dec_start_date AND ret2_dec_milestone_date)) OR (declaration_type = 'retained-3' AND (dec_date_short NOT BETWEEN ret3_dec_start_date AND ret3_dec_milestone_date)) OR (declaration_type = 'retained-4' AND (dec_date_short NOT BETWEEN ret4_dec_start_date AND ret4_dec_milestone_date)) OR (declaration_type = 'completed' AND (dec_date_short NOT BETWEEN completed_dec_start_date AND completed_dec_milestone_date)))) OR (schedule_identifier LIKE ANY ('ecf-extended%', 'ecf-reduced%', 'ecf-replacement%') AND ((declaration_type = 'started' AND dec_date_short < started_dec_start_date) OR (declaration_type = 'retained-1' AND dec_date_short < ret1_dec_start_date) OR (declaration_type = 'retained-2' AND dec_date_short < ret2_dec_start_date) OR (declaration_type = 'retained-3' AND dec_date_short < ret3_dec_start_date) OR (declaration_type = 'retained-4' AND dec_date_short < ret4_dec_start_date) OR (declaration_type = 'completed' AND dec_date_short < completed_dec_start_date))))) THEN "in both checks"
```
[the remark this answers is not in the document]
Comment (emiliocampa) [reply definitions/marts/looker_studio_marts/ecf/ls_cohort_corruption_checker.sqlx:140]: yes

**Review by consult1815 — APPROVED** [review]

**Pull request merged**